katsdpcontroller/generator.py

```diff
@@ -461,6 +407,9 @@ def _make_fgpu(
         Sensor(int, f"{stream.name}-n-samples-between-spectra",
                "Number of samples between spectra",
                default=stream.n_samples_between_spectra, initial_status=Sensor.Status.NOMINAL),
+        Sensor(int, f"{stream.name}-pfb-group-delay",
+               "PFB group delay, specified in number of samples",
+               default=-stream.n_chans * stream.pfb_taps, initial_status=Sensor.Status.NOMINAL),
```

Comment:
I'm still not convinced that this is a correct use of the term "group delay" which as far as I understand it, means the time delay of a signal through a device as a function of frequency. But if that's the term used higher up on the system level then there's little point I guess in being stubborn about it.

I would just want to make sure that `n_chans * pfb_taps` is actually the way that it's calculated on a system level, I've never actually gotten round to getting clarity on that.

Reply:
I've done some tracing through the various tickets, and the goal seems to be that if you have an impulse at timestamp 0, what is the output timestamp of the centroid of the response (modulo the negative sign). Since the PFB filter is symmetric and of length n_chans * n_taps * 2, the centroid is at n_chans * n_taps (actually I think it should be n_chans * n_taps - 0.5, but it's specified to be integer). I've also confirmed that this is what MeerKAT is exposing in a few different instruments.

Reply:
To add to the confusion... :grin:

"Group delay" is the local phase slope at a given frequency. It can therefore vary across the band if you have a wobbly phase response.
"Phase delay" is the "global" phase slope obtained by connecting the phase response at a given frequency with the origin.

Since the PFB is a FIR filter, it has linear phase. This makes the above distinctions moot, since the phase response is a straight line through the origin. Therefore, group delay = phase delay = "delay".